### PR TITLE
Fix error when MA shutsdown

### PR DESCRIPTION
**Pull request by teancom**
Two issues in mass.py:

1. stop() called session.detach() which sets session._connector to None,
    then checked session.connector to close it — so the connector was
    never actually closed. Use session.close() which properly awaits
    connector.close() before clearing the reference.

2. _register_api_commands used getattr() on all attributes, triggering
    lazy property initialization (e.g. http_session) and creating an
    aiohttp ClientSession/connector as a side effect. Skip properties
    during registration since API commands are always methods.

Example log:

```
2026-03-05 16:37:33.223 DEBUG (MainThread) [music_assistant.controllers.config] Stopped.
2026-03-05 16:37:33.281 ERROR (MainThread) [music_assistant] Error doing job: Unclosed connector:   File "/app/venv/bin/mass", line 10, in <module>
    sys.exit(main())
  File "/app/venv/lib/python3.13/site-packages/music_assistant/__main__.py", line 254, in main
    run(
  File "/app/venv/lib/python3.13/site-packages/aiorun/__init__.py", line 257, in run
    loop.run_forever()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 683, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.13/asyncio/base_events.py", line 2042, in _run_once
    handle._run()
  File "/usr/local/lib/python3.13/asyncio/events.py", line 89, in _run
    self._context.run(self._callback, *self._args)
  File "/app/venv/lib/python3.13/site-packages/aiorun/__init__.py", line 223, in new_coro
    await coro
  File "/app/venv/lib/python3.13/site-packages/music_assistant/__main__.py", line 248, in start_mass
    await mass.start()
  File "/app/venv/lib/python3.13/site-packages/music_assistant/mass.py", line 205, in start
    self._register_api_commands()
  File "/app/venv/lib/python3.13/site-packages/music_assistant/mass.py", line 806, in _register_api_commands
    obj = getattr(cls, attr_name)
  File "/app/venv/lib/python3.13/site-packages/music_assistant/mass.py", line 279, in http_session
    self._http_session = create_clientsession(self, verify_ssl=True)
  File "/app/venv/lib/python3.13/site-packages/music_assistant/helpers/aiohttp_client.py", line 43, in create_clientsession
    connector=_get_connector(mass, verify_ssl),
  File "/app/venv/lib/python3.13/site-packages/music_assistant/helpers/aiohttp_client.py", line 174, in _get_connector
    return MusicAssistantTCPConnector(
  File "/app/venv/lib/python3.13/site-packages/aiohttp/connector.py", line 993, in __init__
    super().__init__(
```

### CODE REVIEW

**Review by marcelveldt-traveling — APPROVED** [review]

**Review by marcelveldt-traveling — APPROVED** [review]

**Review by MarvinSchenkel — APPROVED** [review]
Look good, thanks @teancom !
